src/H5Ppublic.h

```diff
@@ -6911,16 +6907,545 @@ H5_DLL herr_t H5Pset_mcdt_search_cb(hid_t plist_id, H5O_mcdt_search_cb_t func, v
 #define H5P_NO_CLASS H5P_ROOT
 
 /* Typedefs */
+/**
+ * \ingroup GPLOA
+ *
+ * \brief Registers a permanent property with a property list class
+ *
+ * \plistcls_id{cls_id}
+ * \param[in] name       Name of property to register
+ * \param[in] size       Size of property in bytes
+ * \param[in] def_value  Default value for property in newly created
+ *                       property lists
+ * \param[in] prp_create Callback routine called when a property list is
+ *                       being created and the property value will be
+ *                       initialized
+ * \param[in] prp_set    Callback routine called before a new value is
+ *                       copied into the property's value
+ * \param[in] prp_get    Callback routine called when a property value is
+ *                       retrieved from the property
+ * \param[in] prp_del    Callback routine called when a property is deleted
+ *                       from a property list
+ * \param[in] prp_copy   Callback routine called when a property is copied
+ *                       from a property list
+ * \param[in] prp_close  Callback routine called when a property list is
+ *                       being closed and the property value will be
+ *                       disposed of
+ *
+ * \return  \herr_t
+ *
+ * \details H5Pregister1() registers a new property with a property list
+ *          class. The property will exist in all property list objects
+ *          of that class after this routine is finished.  The name of
+ *          the property must not already exist.  The default property
+ *          value must be provided and all new property lists created
+ *          with this property will have the property value set to the
+ *          default provided.  Any of the callback routines may be set
+ *          to NULL if they are not needed.
+ *
+ *          Zero-sized properties are allowed and do not store any data in
+ *          the property list. These may be used as flags to indicate the
+ *          presence or absence of a particular piece of information. The
+ *          default pointer for a zero-sized property may be set to NULL.
+ *          The property \p prp_create and \p prp_close callbacks are called for
+ *          zero-sized properties, but the \p prp_set and \p prp_get callbacks are
+ *          never called.
+ *
+ *          The \p prp_create routine is called when a new property list with
+ *          this property is being created. The #H5P_prp_create_func_t
+ *          callback function is defined as follows:
+ *
+ *          \snippet this H5P_prp_cb1_t_snip
```

Comment:
Using snippets for callbacks is the way to go. Great! But we shouldn't repeat the parameter lists everywhere. Since Doxygen hyperlinks everything, the user has to just click on the callback type and will be taken to the documentation. The parameters should be defined once with the callback. As an example see `H5E_walk2_t` in `H5Epublic.h` around line 159. Does that make sense?